### DISCUSSION TITLE
Graph of transitive dependencies

### DISCUSSION THREAD

**Discussion by marcosh** [flora-pm/flora-server#397] [Feature Requests]
Currently flora provides information about the dependency and the dependents of every single package.

It would be nice to know whether a package A depends somehow on a package B, meaning that there is a chain of dependencies connecting A and B.

I'm not sure what would be the best way to visualise such information. Some potential options:
- visualise all the possible dependency paths between A and B. Could be just a list of strings, where every string is a dependency path from A to B. Could be a directed graph showing all the paths from A to B
- visualise all the transitive dependencies of `A`
- visualise all the transitive dependents of `B`

I don't know the internals of flora and, in particular, I don't know how the dependency information are stored.

Probably the best way to retrieve the data about dependency graphs would be to store at least some data in a graph database and use to query all the paths from `A` to `B`.

**Comment by tchoutri** [comment]
Thankfully the database engine is perfectly capable of performing such queries. :)

**Reply by marcosh** [reply]
awesome, so this is not an issue
